src/main/java/ca/team2706/frc/robot/config/Config.java

```diff
@@ -70,6 +84,43 @@ public static void initialize() {
         }
     }
 
+    /**
+     * Reads the robot type from the filesystem
+     *
+     * @return The integer ID of the robot defaulting to 0
+     */
+    private static int getRobotId() {
+        int id = 0;
+
+        try (BufferedReader reader = Files.newBufferedReader(ROBOT_ID_LOC)) {
+            id = Integer.parseInt(reader.readLine());
+        } catch (IOException | NumberFormatException e) {
+            id = 0;
+            e.printStackTrace();
+        }
+
+        return id;
+    }
+
+    /**
+     * Returns one of the values passed based on the robot ID
+     *
+     * @param first The first value (default value)
+     * @param more Other values that could be selected
+     * @param <T> The type of the value
+     * @return The value selected based on the ID of the robot
+     */
+    @SafeVarargs
+    private static <T> T robotSpecific(T first, T... more) {
+        // Return the first value if the robot id doesn't fall between second and last index
+        if(ROBOT_ID < 1 || ROBOT_ID > more.length) {
+            return first;
+        }
+        else {
+            return more[ROBOT_ID - 1];
+        }
+    }

```

Comment:
More of a question here, but why didn't you just do

```java
private static <T> T robotSpecific(T... configurations) {
}
```

Reply:
Normally it would be fine. I just wanted it so that there has to be at least one value passed, and not just an empty list. Then I can guarantee that one of the values passed will be returned.

Reply:
Makes sense. Thanks!